### PR TITLE
envoy/eds: only respond to requested clusters

### PR DESCRIPTION
**Pull request by shashankram**
**Description**:
Optimizes EDS to only respond with the endpoints for
requested clusters. Per the XDS protocol, additional
responses for resources that have not been requested
will be ignored.

Signed-off-by: Shashank Ram <shashr2204@gmail.com>

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| Control Plane              | [ X] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project? `no`
    -   Did you notify the maintainers and provide attribution?

1. Is this a breaking change? `no`

### CODE REVIEW

**Comment by eduser25** [comment]
This is good

File: pkg/envoy/eds/response.go [hunk 1]
```diff
@@ -1,55 +1,59 @@
...
-	allowedServicesEndpoints := make(map[service.MeshService][]endpoint.Endpoint)
-
-	for _, dstSvc := range meshCatalog.ListAllowedOutboundServicesForIdentity(proxyIdentity) {
-		endpoints, err := meshCatalog.ListAllowedEndpointsForService(proxyIdentity, dstSvc)
-		if err != nil {
-			log.Error().Err(err).Msgf("Failed listing allowed endpoints for service %s for proxy identity %s", dstSvc, proxyIdentity)
-			continue
-		}
-		allowedServicesEndpoints[dstSvc] = endpoints
+func clusterToMeshSvc(cluster string) (service.MeshService, error) {
```
Comment (eduser25) [inline pkg/envoy/eds/response.go:53]: would it make sense to have a generic `service.GetMeshServiceFromString`? not sure other parts of the code would need it

**Review by snehachhabria — APPROVED** [review]

File: pkg/envoy/sds/response.go [hunk 2]
```diff
@@ -83,6 +83,7 @@ func (s *sdsImpl) getSDSSecrets(cert certificate.Certificater, requestedCerts []
 					requestedCertificate, proxy.GetCertificateSerialNumber(), proxy.GetPodUID())
 				continue
 			}
+			log.Debug().Msgf("Successfully created SDS secret %s for proxy %s", envoySecret, proxy.GetCertificateCommonName())
```
Comment (eduser25) [inline pkg/envoy/sds/response.go:86]: do you want this in this commit?

**Review by eduser25 — APPROVED** [review]
good stuff, seal of approval

File: pkg/envoy/eds/response.go (hunk 1 above)
Comment (shashankram) [reply pkg/envoy/eds/response.go:53]: Right, I don't see a use case to make it that generic atm.

File: pkg/envoy/sds/response.go (hunk 2 above)
Comment (shashankram) [reply pkg/envoy/sds/response.go:86]: Will drop this, thanks!

**Pull request merged**